lib/fizzy/execute.cpp

```diff
@@ -1437,4 +1448,74 @@ std::optional<FuncIdx> find_exported_function(const Module& module, std::string_
 
     return {};
 }
+
+std::optional<ExternalGlobal> find_exported_global(Instance& instance, std::string_view name)
+{
+    const auto opt_index = find_export(instance, ExternalKind::Global, name);
+    if (!opt_index.has_value())
+        return std::nullopt;
+
+    const auto global_idx = *opt_index;
+    if (global_idx < instance.imported_globals.size())
+    {
+        // imported global is reexported
+        return ExternalGlobal{instance.imported_globals[global_idx].value,
+            instance.imported_globals[global_idx].is_mutable};
+    }
+    else
+    {
+        // global owned by instance
+        return ExternalGlobal{
+            &instance.globals[global_idx], instance.module.globalsec[global_idx].is_mutable};
+    }
+}
+
+std::optional<ExternalTable> find_exported_table(Instance& instance, std::string_view name)
+{
+    if (!find_export(instance, ExternalKind::Table, name))
+        return std::nullopt;
+
+    if (instance.module.tablesec.size() == 1)
+    {
+        // table owned by instance
+        return ExternalTable{instance.table.get(), instance.module.tablesec[0].limits};
+    }
+    else
+    {
+        // imported table is reexported
+        const auto it_import =
+            std::find_if(instance.module.importsec.begin(), instance.module.importsec.end(),
+                [](const auto& import) { return import.kind == ExternalKind::Table; });
+        assert(it_import != instance.module.importsec.end());
+
+        return ExternalTable{instance.table.get(), it_import->desc.table.limits};
```

Comment:
Limits here are not exactly correct: table could have been imported with limits narrower than the ones defined in import's module, we don't save those during instantiate.

Reply:
Similar for memory, but there we save at least the provided upper limit (in `Instance::memory_max_pages`), I use it below in `find_exported_memory`

Reply:
Can you add (possible disabled tests) for these?

Reply:
Added FIXME comments and disabled tests for table and memory cases of this.